src/core/qgsmaplayer.cpp

```diff
@@ -1385,51 +1385,84 @@ QString QgsMapLayer::saveNamedStyle( const QString &uri, bool &resultFlag, Style
   return saveNamedProperty( uri, QgsMapLayer::Style, resultFlag, categories );
 }
 
-void QgsMapLayer::exportSldStyle( QDomDocument &doc, QString &errorMsg ) const
+void QgsMapLayer::exportSldStyle( QDomDocument &doc, QString &errorMsg )
```

Comment:
Could you leave this as const, but make the edited members mutable?

Reply:
see above